addons/bus/models/ir_websocket.py

```diff
@@ -20,6 +20,8 @@ def _build_bus_channel_list(self, channels):
         return channels
 
     def _subscribe(self, data):
+        if [c for c in data['channels'] if not isinstance(c, str)]:
+            raise ValueError("bus.Bus only string channels are allowed.")
```

Comment:
Why here and not above? It's usually easier to follow if params are validated close where received.

Reply:
Because we want to validate the initial channel list (the one coming from the front end). Moreover, why bother constructing the channel list if we know since the beginning that we won't accept it.